### PR TITLE
Support checkstyle XML format as an input

### PR DESCRIPTION
**Pull request by haya14busa**

### CODE REVIEW

File: parser_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	}
+
+	p := NewCheckStyleParser()
+	crs, err := p.Parse(strings.NewReader(sample))
+	if err != nil {
+		t.Error(err)
+	}
+	for i, cr := range crs {
+		if got, want := cr.Lines[0], wants[i]; got != want {
+			t.Error("%d: got %v, want %v", i, got, want)
```
Comment (reviewdog) [inline parser_test.go]: **[govet]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
possible formatting directive in Error call

File: parser.go [hunk 2]
```diff
@@ -35,3 +37,55 @@ func (p *ErrorformatParser) Parse(r io.Reader) ([]*CheckResult, error) {
...
+	Version string            `xml:"version,attr"`
+	Files   []*CheckStyleFile `xml:"file,omitempty"`
+}
+
+type CheckStyleFile struct {
+	Name   string             `xml:"name,attr"`
+	Errors []*CheckStyleError `xml:"error"`
+}
+
+type CheckStyleError struct {
```
Comment (reviewdog) [inline parser.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type CheckStyleError should have comment or be unexported

File: parser.go [hunk 3]
```diff
@@ -35,3 +37,55 @@ func (p *ErrorformatParser) Parse(r io.Reader) ([]*CheckResult, error) {
...
+					fmt.Sprintf("%v:%d:%d: %v: %v (%v)",
+						file.Name, cerr.Line, cerr.Column, cerr.Severity, cerr.Message, cerr.Source),
+				},
+			})
+		}
+	}
+	return rs, nil
+}
+
+type CheckStyleResult struct {
```
Comment (reviewdog) [inline parser.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type CheckStyleResult should have comment or be unexported

File: parser.go [hunk 4]
```diff
@@ -35,3 +37,55 @@ func (p *ErrorformatParser) Parse(r io.Reader) ([]*CheckResult, error) {
...
+
+// CheckStyleParser is checkstyle parser.
+type CheckStyleParser struct{}
+
+// NewCheckStyleParser returns a new CheckStyleParser.
+func NewCheckStyleParser() Parser {
+	return &CheckStyleParser{}
+}
+
+func (p *CheckStyleParser) Parse(r io.Reader) ([]*CheckResult, error) {
```
Comment (haya14busa) [inline parser.go]: ack. This is just an implementation of Parse method of Parser interface

**Pull request merged**